gateway-ha/src/main/java/io/trino/gateway/ha/router/HaGatewayManager.java

```diff
@@ -82,13 +83,33 @@ public Optional<ProxyBackendConfiguration> getBackendByName(String name)
     @Override
     public void deactivateBackend(String backendName)
     {
-        dao.deactivate(backendName);
+        updateBackendActivationStatus(backendName, false, () -> dao.deactivate(backendName));
     }
 
     @Override
     public void activateBackend(String backendName)
     {
-        dao.activate(backendName);
+        updateBackendActivationStatus(backendName, true, () -> dao.activate(backendName));
+    }
+
+    private void updateBackendActivationStatus(String clusterName, boolean newStatus, Runnable action)
+    {
+        GatewayBackend model = dao.findFirstByName(clusterName);
+        if (model == null) {
+            throw new IllegalStateException("No cluster found with name: " + clusterName + ", could not (de)activate");
+        }
+
+        Boolean prevStatus = model.active();
+        action.run();
+        logActivationStatusChange(clusterName, newStatus, prevStatus);
+    }
+
+    private void logActivationStatusChange(String clusterName, boolean newStatus, boolean prevStatus)
+    {
+        if (!Objects.equals(prevStatus, newStatus)) {
```

Comment:
Since `newStatus`/`prevStatus` are primitive type (`boolean`), you can just do `if (prevStatus != newStatus)`. We only need `Objects.equals()` if we use the boxed type (`Boolean`)